src/Orleans.Runtime/Configuration/Options/GrainCollectionOptions.cs

```diff
@@ -25,5 +25,8 @@ public class GrainCollectionOptions
         /// Period of inactivity necessary for a grain to be available for collection and deactivation by grain type.
         /// </summary>
         public Dictionary<string, TimeSpan> ClassSpecificCollectionAge { get; set; } = new Dictionary<string, TimeSpan>();
+
+        public long CollectionMemoryThresholdInMegabytes { get; set; } = DEFAULT_COLLECTION_MEMORY_THRESHOLD_IN_MEGABYTES;
```

Comment:
This needs a doc comment to explain the behavior.

When this value is non-zero, collection will be suspended until memory usage passes the specified threshold.

Additionally, does this force early collection if the memory usage is above the threshold? I believe that's the "least surprises" behavior.

Reply:
The implementation is very simple at this point. I'll add some documentation, that was a stupid oversight. Right now it basically uses the instrumentation that is already in place to decide whether or not to do a collection of grains at all, and if it decides to do so it does so using the existing logic where it follows the normal rules of Orleans grain collection. That is, given that the current memory usage as returned by GC.GetTotalMemory() is greater than the configured threshold, it will run collection as usual, if not it will simply suppress collection.